### PR TITLE
[nats-streaming] Release v0.17.0

### PR DESCRIPTION
**Pull request by kozlovic**
Details can be found [here](https://github.com/nats-io/nats-streaming-server/releases/tag/v0.17.0)

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

**Comment by kozlovic** [comment]
Related doc PR: https://github.com/docker-library/docs/pull/1654

File: library/nats-streaming [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 
-Tags: 0.16.2-nanoserver, nanoserver, nanoserver-1803
-SharedTags: 0.16.2, latest
+Tags: 0.17.0-nanoserver, nanoserver, nanoserver-1803
+SharedTags: 0.17.0, latest
 Architectures: windows-amd64
-windows-amd64-GitCommit: 236d21372934860eb197cc4b25ad7e58b981ed6f
+windows-amd64-GitCommit: 9716618a61deba8a7e28b69d53d95d698bde5117
 windows-amd64-Directory: windows/nanoserver-1803
 Constraints: nanoserver-1803
```
Comment (tianon) [inline library/nats-streaming]: This whole section should be removed, right? (https://github.com/nats-io/nats-streaming-docker/commit/a13ea6748fdfc3de616619ffb3b2d3b842bd1fea)

Comment (kozlovic) [reply library/nats-streaming]: So sorry... I was going to, got distracted and pushed before removing. I did amend the commit and "push force".

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 29c7b0c..d527a71 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -2,6 +2,5 @@ nats-streaming:linux @ amd64
 nats-streaming:linux @ arm32v6
 nats-streaming:linux @ arm32v7
 nats-streaming:linux @ arm64v8
-nats-streaming:nanoserver-1803 @ windows-amd64
 nats-streaming:nanoserver-1809 @ windows-amd64
 nats-streaming:windowsservercore @ windows-amd64
diff --git a/_bashbrew-list b/_bashbrew-list
index 81fa781..44170e3 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,11 +1,8 @@
-nats-streaming:0.16.2
-nats-streaming:0.16.2-linux
-nats-streaming:0.16.2-nanoserver
-nats-streaming:0.16.2-nanoserver-1809
-nats-streaming:0.16.2-windowsservercore
+nats-streaming:0.17.0
+nats-streaming:0.17.0-linux
+nats-streaming:0.17.0-nanoserver-1809
+nats-streaming:0.17.0-windowsservercore
 nats-streaming:latest
 nats-streaming:linux
-nats-streaming:nanoserver
-nats-streaming:nanoserver-1803
 nats-streaming:nanoserver-1809
 nats-streaming:windowsservercore
diff --git a/_bashbrew-shared-tags b/_bashbrew-shared-tags
index f83c306..d602d5e 100644
--- a/_bashbrew-shared-tags
+++ b/_bashbrew-shared-tags
@@ -1,2 +1,2 @@
-nats-streaming:0.16.2 -- nats-streaming:linux, nats-streaming:nanoserver-1803
-nats-streaming:latest -- nats-streaming:linux, nats-streaming:nanoserver-1803
+nats-streaming:0.17.0 -- nats-streaming:linux
+nats-streaming:latest -- nats-streaming:linux
diff --git a/nats-streaming_linux/nats-streaming-server b/nats-streaming_linux/nats-streaming-server
index 2bf66df..dd0ef07 100755
Binary files a/nats-streaming_linux/nats-streaming-server and b/nats-streaming_linux/nats-streaming-server differ
diff --git a/nats-streaming_nanoserver-1803/Dockerfile b/nats-streaming_nanoserver-1803/Dockerfile
deleted file mode 100644
index 26c3752..0000000
diff --git a/nats-streaming_nanoserver-1803/nats-streaming-server.exe b/nats-streaming_nanoserver-1803/nats-streaming-server.exe
deleted file mode 100755
index 84f2ebe..0000000
diff --git a/nats-streaming_nanoserver-1809/nats-streaming-server.exe b/nats-streaming_nanoserver-1809/nats-streaming-server.exe
index 84f2ebe..1f3dc70 100755
Binary files a/nats-streaming_nanoserver-1809/nats-streaming-server.exe and b/nats-streaming_nanoserver-1809/nats-streaming-server.exe differ
diff --git a/nats-streaming_windowsservercore/nats-streaming-server.exe b/nats-streaming_windowsservercore/nats-streaming-server.exe
index 84f2ebe..1f3dc70 100755
Binary files a/nats-streaming_windowsservercore/nats-streaming-server.exe and b/nats-streaming_windowsservercore/nats-streaming-server.exe differ
```

</details>

**Comment by tianon** [comment]
LGTM

Build test of #7453; be5b5803; `amd64` (`nats-streaming`):

```console

$ bashbrew build nats-streaming:0.17.0-linux
Building bashbrew/cache:2a107f9779ec7a53276a5edbfe1fc36755d90505e01e009d532d1b9f37dbc641 (nats-streaming:0.17.0-linux)
Tagging nats-streaming:0.17.0-linux
Tagging nats-streaming:linux
Tagging nats-streaming:0.17.0
Tagging nats-streaming:latest

$ test/run.sh nats-streaming:0.17.0-linux
testing nats-streaming:0.17.0-linux
	'cve-2014--shellshock' [1/2]...passed
	'override-cmd' [2/2]...image does not appear to contain "echo" -- assuming single-binary image
passed


$ bashbrew build nats-streaming:0.17.0-nanoserver-1809
skipping "nats-streaming:0.17.0-nanoserver-1809" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh nats-streaming:0.17.0-nanoserver-1809
testing nats-streaming:0.17.0-nanoserver-1809
	image has no tests...skipping


$ bashbrew build nats-streaming:0.17.0-windowsservercore
skipping "nats-streaming:0.17.0-windowsservercore" (due to architecture "amd64"; only "windows-amd64" supported)

$ test/run.sh nats-streaming:0.17.0-windowsservercore
testing nats-streaming:0.17.0-windowsservercore
	image has no tests...skipping

```

**Comment by kozlovic** [comment]
@tianon I know that this is a common problem, but some of the images (linux amd64 and windows) are still not avail. Is that normal?
```
ivan$ docker run --rm -ti mplatform/mquery nats-streaming:0.17.0
Image: nats-streaming:0.17.0
 * Manifest List: Yes
 * Supported platforms:
   - linux/arm/v6
   - linux/arm/v7
   - linux/arm64
```

**Comment by tianon** [comment]
Everything looks green under https://doi-janky.infosiftr.net/job/multiarch/view/images/view/nats-streaming/ so this needs more digging for amd64 at least -- you don't have `windows` because the `SharedTags` that had `windows` were deleted in this PR, and should probably be added back on a different version (or on all Windows versions).

**Comment by tianon** [comment]
I think `amd64` had a fluke in pushing -- I've restarted it and it pushed this time, so the manifest list under `library` should be updated shortly.

**Comment by kozlovic** [comment]
@tianon 

> you don't have windows because the SharedTags that had windows were deleted in this PR, and should probably be added back on a different version (or on all Windows versions).

That's a bummer.. yes, when removing the no longer supported nanoserver-1803, that removed the "SharedTags" line.. 

If I submit a new PR with that added, but everything the same, will that work? In other words, I hope I don't need to do a new version of the streaming server just for that, right?

**Comment by tianon** [comment]
No need to do a new version -- adding SharedTags in a new PR should do the
right thing.

**Comment by kozlovic** [comment]
@tianon Thanks, I have submitted PR: https://github.com/docker-library/official-images/pull/7473 to fix that. And yes, I see the amd64 image now in the manifest. Thanks!

**Pull request merged**